llap-tez/src/test/org/apache/hadoop/hive/llap/tezplugins/TestLlapTaskSchedulerService.java

```diff
@@ -1115,7 +1116,7 @@ public void testHostPreferenceMissesConsistentPartialAlive() throws IOException,
       // 3rd task requested host3, got host1 since host3 is dead and host4 is full
       assertEquals(HOST1, argumentCaptor2.getAllValues().get(2).getNodeId().getHost());
 
-      verify(tsWrapper.mockServiceInstanceSet, times(2)).getAllInstancesOrdered(true);
+      verify(tsWrapper.mockServiceInstanceSet, atLeast(2)).getAllInstancesOrdered(true);
```

Comment:
Why was this converted to atLeast instead of times? The test seems pretty deterministic.

Reply:
Good catch, the main reason this was converted to atLeast is that before: getAllInstancesOrdered was only called when a Task had to rollover to the next node when that was disabled as in the test above.
However now, on every scheduler call we get the Alive hosts in-order (using getAllInstancesOrdered when using consistent hashing) -- the problem here is we dont know how many time the scheduling loop will be called when the test finishes -- thus changed to at least (2) needed for the requests to make progress.